src/librustc/util/ppaux.rs

```diff
@@ -709,6 +709,9 @@ define_print! {
 
 define_print! {
     ('tcx) ty::ExistentialTraitRef<'tcx>, (self, f, cx) {
+        display {
+            cx.parameterized(f, self.substs, self.def_id, &[])
```

Comment:
This looks wrong. The `substs` are missing the `Self` type, which is why the `debug {...}` implementation below does through all of that trouble of creating a `TraitRef`.

I think the correct fix is moving the code from `debug {...}` to `display {...}` and changing `debug {...}` to contain just `self.print_display(f, cx)`.